### PR TITLE
feat: Support to use GeneratorFunction as `options.origin`

### LINKED ISSUE #8: Support to set GeneratorFunction as `option.origin`
I want to config `Access-Control-Allow-Origin` dynamically, while acquiring the configuration is an asynchronous operation.

### PR DESCRIPTION
**Pull request by chinesedfan**
Closed #8.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,6 +61,9 @@ module.exports = function (options) {
     var origin;
     if (typeof options.origin === 'function') {
       origin = options.origin(this);
```
Comment (fengmk2) [inline index.js]: check `options.origin.constructor.name` will be better.

```js
if (typeof options.origin === 'function') {
  if (options.origin.constructor.name === 'GeneratorFunction') {
    origin = yield options.origin(this);
  } else {
    origin = options.origin(this);
  }
}
```

**Comment by fengmk2** [comment]
@chinesedfan Can you also pick this feature into 2.x branch? https://github.com/koajs/cors/tree/v2.x

**Comment by chinesedfan** [comment]
I have updated the way to check.

But in v2.x branch, does it mean `async/await` should be introduced?

**Comment by fengmk2** [comment]
Landed 32d9197e75ce344e8c515ddcb9660de2ecd75635

**Pull request closed without merging**